rules/prefer-t-regex.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+const {visitIf} = require('enhance-visitors');
+const createAvaRule = require('../create-ava-rule');
+const util = require('../util');
+
+const create = context => {
+	const ava = createAvaRule();
+
+	const booleanTests = [
+		'true',
+		'false',
+		'truthy',
+		'falsy'
+	];
+
+	const findReference = name => {
+		const reference = context.getScope().references.find(ref => ref.identifier.name === name);
+		const definitions = reference.resolved.defs;
+		return definitions[definitions.length - 1].node;
+	};
+
+	return ava.merge({
+		CallExpression: visitIf([
+			ava.isInTestFile,
+			ava.isInTestNode
+		])(node => {
+			// Call a boolean assertion (eg: `.true`, `.false` ...)
+			if (node.callee.type === 'MemberExpression' &&
+				booleanTests.includes(node.callee.property.name) &&
+				util.getNameOfRootNodeObject(node.callee) === 't') {
```

Comment:
The above expression could be moved into a descriptive variable, like `isBooleanAssertion`. Then you can use early-return instead to reduce the nesting.